internal/validate/validate.go

```diff
@@ -0,0 +1,29 @@
+package validate
+
+import (
+	"context"
+	"errors"
+
+	platformv1alpha1 "github.com/openshift/api/platform/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	errNotUniquePackage = errors.New("spec.package.name is not unique")
```

Comment:
Do we need to perform any errors.Is operations here? Might be too premature. Not too sure if exported/unexpected makes a difference.

Reply:
I was going to use it in our E2E tests so I can easily target it but I get what you're saying here. Its probably premature though.

Reply:
Yeah I think we can just blindly process the error returned by this function at the call site without the want/need to parse based on the error type returned.